### PR TITLE
Docs: readme creative updates

### PR DESCRIPTION
**Pull request by mnaumanali94**
Hey @XVincentX,

Made a few upgrades to the readme by adding visuals and a few subtle references to Stoplight. It also adds contract testing as a major use case in the header. @marbemac suggested we update our docs to write more about the contract testing use case and link it here.

Let me know what you think. Happy to make changes and help with docs.

### CODE REVIEW

**Review by XVincentX — CHANGES_REQUESTED** [review]
I am not really ok with these changes.

While I do understand the intentions, I do observe that:

* The foreword of the readme has been completely replaced by an image, even removing the initial title. Apart from the SEO (the content is not indexable), generally speaking this makes it even unreadable from a terminal (yes, there are people doing that).

image

Now that I see what we have, I also think removing the CircleCI links or at least put them as a reference down in the document is better.

* While I personally do not like this spread idea that emoji should be everywhere — since they're unicode characters and can be rendered correctly from a terminal, I can live with that

* I see an eccessive number of links to Stoplight that are really unnecessary. We already have a number of references, and I feel we should not go over that.

* I do not understand what kind of pull request should somebody create in case if they have an interesting use case. Feel like a buzzword with no actual call to action

* There's already a License file — I do not see the reason to link it from the README

* I understand the plant a tree thing, but using an image taking the entire screen is a bit too much

---

Bottom line: the README should be nicely viewable both as plain text and as a rendered version. That is what markdown is about. This PR is leaning this a little bit too much on the other side

**Comment by mnaumanali94** [comment]
@XVincentX Thanks for the review! Made a few changes based on the comments:

1. Removed the HTML to use markdown only

2. Added an alt text to the header. Moreover, the image used is an SVG so its indexable by search engines. See how it looks now in the terminal:
Screenshot 2020-12-17 at 7 28 08 PM

3. Removed the license file and picture from Ecologi (Agree, it was too much)

4. For the use case, removed the pull request part. What we are looking to do is get a few case studies for prism to put in a list here much like the [list of people using OpenAPI code generator](https://github.com/OpenAPITools/openapi-generator#4---companiesprojects-using-openapi-generator)

5. Added an overview section as that is what shows up as the heading in table of contents in Stoplight. Also, this should make it a little easier to navigate through the readme on GitHub.

6. For CircleCI link, you mean the build status? That's a common practice i believe.

7. The Stoplight links i added should mostly be useful for people. Like somebody looking for a hosted version or Studio's integration. Any particular links that you think we should remove?

P.S. This is the analog that I mostly used for some of the design changes: https://github.com/segmentio/evergreen

File: README.md [hunk 1]
```diff
@@ -1,50 +1,70 @@
-# Prism
```
Comment (XVincentX) [inline README.md:1]: I think the main title should still stay

File: README.md [hunk 2]
```diff
@@ -1,50 +1,70 @@
...
+[![Prism is a set of packages for API mocking with OpenAPI v2 (formerly known as Swagger) and OpenAPI v3.](./examples/readme-header.svg)](https://stoplight.io/api-mocking?utm_source=github&utm_medium=prism&utm_campaign=readme)
```
Comment (XVincentX) [inline README.md:1]: I really do not see the value of putting an image. I still vote to keep it in the way it is.

Comment (philsturgeon) [inline README.md:1]:
```suggestion
[![Prism is a set of packages for API mocking with OpenAPI v2 (formerly known as Swagger) and OpenAPI v3.x](./examples/readme-header.svg)](https://stoplight.io/api-mocking?utm_source=github&utm_medium=prism&utm_campaign=readme)
```

File: README.md [hunk 3]
```diff
@@ -1,50 +1,70 @@
...
 
 [![CircleCI](https://img.shields.io/circleci/build/github/stoplightio/prism/master)](https://circleci.com/gh/stoplightio/prism)
```
Comment (XVincentX) [inline README.md:3]: (Also referring to your previous comment) I think a better idea would be to use deferred links, such as

[Link][1]

```markdown
[Link][1]

[1]: http://google.it
```

so that we clean up this section from the long strings.

File: README.md [hunk 4]
```diff
@@ -1,50 +1,70 @@
...
 
+**Installation**
 ```bash
 npm install -g @stoplight/prism-cli
 
 # OR
 
 yarn global add @stoplight/prism-cli
 ```
+> Prism requires NodeJS >= 12 to properly work.
```
Comment (XVincentX) [inline README.md:33]: Why is this quoted?

File: README.md [hunk 5]
```diff
@@ -92,3 +127,7 @@ Check these projects out!
 [cli-docs]: ./docs/getting-started/03-cli.md
 [2.x]: https://github.com/stoplightio/prism/tree/2.x
 [http-docs]: packages/http/README.md
+
+## 🌲 Sponsor Prism by Planting a Tree
+
+If you would like to thank us for creating prism, we ask that you [**buy the world a tree**](https://offset.earth/stoplightinc).
```
Comment (philsturgeon) [inline README.md:133]:
```suggestion
If you would like to thank us for creating prism, we ask that you [**buy the world a tree**](https://ecologi.com/stoplightinc).
```

**Review by philsturgeon — APPROVED** [review]
Awesome again!

File: README.md (hunk 2 above)
Comment (XVincentX) [reply README.md:1]: Moreover, in this way the text is hidden from the image, in case the download succeeds — not too sure this is what we want.

File: README.md [hunk 6]
```diff
@@ -1,50 +1,70 @@
...
+[![Prism is a set of packages for API mocking with OpenAPI v2 (formerly known as Swagger) and OpenAPI v3.](./examples/readme-header.svg)](https://stoplight.io/api-mocking?utm_source=github&utm_medium=prism&utm_campaign=readme)
 
 [![CircleCI](https://img.shields.io/circleci/build/github/stoplightio/prism/master)](https://circleci.com/gh/stoplightio/prism)
 [![NPM Downloads](https://img.shields.io/npm/dw/@stoplight/prism-http?color=blue)](https://www.npmjs.com/package/@stoplight/prism-cli)
 [![Buy us a tree](https://img.shields.io/badge/Buy%20us%20a%20tree-%F0%9F%8C%B3-lightgreen)](https://offset.earth/stoplightinc)
 
-Prism is a set of packages for API mocking with **OpenAPI v2** (formerly known as Swagger) and **OpenAPI v3**.
-
-For example, given an API description document you can spin up a mock HTTP server and respond realistically based on your requests
+- **Mock Servers**: Life-like mock servers from any API Specification Document.
```
Comment (XVincentX) [inline README.md:7]: I feel we're missing a section — we're jumping directly to the main features without introducing them. The above sentence covered by the image would do such job.

File: README.md [hunk 7]
```diff
@@ -1,50 +1,70 @@
...
 > Note: This branch refers to Prism 3.x, which is the current version most likely you will use. If you're looking for the 2.x version, look at the [`2.x` branch][2.x]
 
-## Installation
+# Overview
+
+- [Installation and Usage](#-installation-and-Usage)
+- [Documentation and Community](#-documentation-and-community)
+- [Roadmap](#-roadmap)
+- [FAQs](#-faqs)
+- [Contributing](#-contributing)
```
Comment (XVincentX) [inline README.md:21]: I suggest to remove this, since keeping this up to date is something nobody will ever do (see platform internal) and links will point to the empty space.

A smart renderer should do this, not us directly in the document.

File: README.md [hunk 8]
```diff
@@ -1,50 +1,70 @@
...
 ```bash
 npm install -g @stoplight/prism-cli
 
 # OR
 
 yarn global add @stoplight/prism-cli
 ```
+> Prism requires NodeJS >= 12 to properly work.
+
+For more installation options, see our [installation documentation](https://meta.stoplight.io/docs/prism/docs/getting-started/01-installation.md).
```
Comment (XVincentX) [inline README.md:35]: This forces me to go on the web when I have all the documentation right in a directory. I'd say to preserve the local link.

File: README.md [hunk 9]
```diff
@@ -92,3 +127,7 @@ Check these projects out!
 [cli-docs]: ./docs/getting-started/03-cli.md
 [2.x]: https://github.com/stoplightio/prism/tree/2.x
 [http-docs]: packages/http/README.md
+
+## 🌲 Sponsor Prism by Planting a Tree
```
Comment (XVincentX) [inline README.md:131]: We already have a button above, do we need to repeat ourselves here?

**Review by XVincentX — COMMENTED** [review]
It'd be great if we could chat a little bit more about the bigger picture of what we want to do here.

File: README.md (hunk 3 above)
Comment (mnaumanali94) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (mnaumanali94) [reply README.md]: I have updated to add a description followed by the image. That way even if the image doesn't load we'll have a description.

The image is for aesthetic purposes and makes for a good reading for normal use cases i.e. laptop and phone

File: README.md (hunk 6 above)
Comment (mnaumanali94) [reply README.md:9]: Added

File: README.md (hunk 8 above)
Comment (mnaumanali94) [reply README.md]: Fixed

File: README.md (hunk 1 above)
Comment (mnaumanali94) [reply README.md:1]: Screenshot 2020-12-22 at 12 41 35 PM

It looks like this in Stoplight which is weird.

The change makes it look like this:
Screenshot 2020-12-22 at 12 45 38 PM

File: README.md (hunk 7 above)
Comment (mnaumanali94) [reply README.md:23]: Screenshot 2020-12-22 at 12 41 35 PM

It looks like this in Stoplight which is weird. So to tackle this i created an overview section. That also makes navigating the readme easier on GitHub. Didn't add all the sections so there's not an urgent need to keep it up to date.

File: README.md (hunk 1 above)
Comment (XVincentX) [reply README.md:1]: Well ok, but then I'd argue the bug is in the platform and should be fixed.

Comment (mnaumanali94) [reply README.md:1]: Wouldn't say its a bug considering most tools would pick up the first heading.

If you think this is in a good spot, can we merge these changes in and make potential improvements later in another PR?

**Comment by philsturgeon** [comment]
If I can offer an opinion, I think this is a good PR which adds consistency with how things look in Spectral. Previous spectral had a logo and Prism didnt so they were lacking consistency. 

I think the ToC is fine (can be updated automatically with a VS Code plugin) and yeah there are some bugs you're noticing Nauman, can you make sure you make GitHub Issues for them in platform-internal and @ me in them or DM them over so I can chase them through? 

Emojis are everywhere but its a current trend, and we dont wanna be old men shaking their walking sticks at the youths complaining about their skateboards and their hippety hop. Let's just go along with the trend for now. :)

**Review by philsturgeon — APPROVED** [review]

**Comment by mnaumanali94** [comment]
Thanks @philsturgeon! I'll keep an eye out and make issues where required.

@XVincentX Think it's in a good spot to get it merged?
